Stratis.Bitcoin/Logging/LogsExtension.cs

```diff
@@ -7,99 +7,181 @@
 using Stratis.Bitcoin.Configuration;
 using Stratis.Bitcoin.Configuration.Settings;
 using Stratis.Bitcoin.Utilities;
+using NLog;
+using NLog.Config;
+using NLog.Targets;
+using System.Text;
 
 namespace Stratis.Bitcoin.Logging
 {
+    /// <summary>
+    /// Integration of NLog with Microsoft.Extensions.Logging interfaces.
+    /// </summary>
     public static class LogsExtension
     {
-        public static void AddFilters(this ILoggerFactory loggerFactory, LogSettings settings)
+        /// <summary>Width of a column for pretty console/log outputs.</summary>
+        public const int ColumnLength = 16;
+
+        /// <summary>Currently used node's log settings.</summary>
+        private static LogSettings logSettings;
+
+        /// <summary>Currently used data folder to determine path to logs.</summary>
+        private static DataFolder folder;
+
+        /// <summary>
+        /// Initializes application logging.
+        /// </summary>
+        static LogsExtension()
+        {
+            // If there is no NLog.config file, we need to initialize the configuration ourselves.
+            if (LogManager.Configuration == null) LogManager.Configuration = new LoggingConfiguration();
+
+            // Installs handler to be called when NLog's configuration file is changed on disk.
+            LogManager.ConfigurationReloaded += NLogConfigurationReloaded;
```

Comment:
Yes, the most of the code complexity here is because of reloading functionality.

The level of interconnection with loggerFactory you are looking for is simply not there. Whole NLog configuration is a static thing, lives in NLog. The only thing we do here is that we need to initialize it correctly (including when it reloads) using what we have in NodeSettings.